### DISCUSSION TITLE
Only either ET or BT showing but not both. Running on MacOS Ventura on M1 Macbook Air, Aritsan 2.8.0, using Bella EVO 1 Roaster

### DISCUSSION THREAD

**Discussion by Zero518** [artisan-roaster-scope/artisan#1053] [Q&A] [answered]
Have an issue with artisan on Macbook Air M1, running version 2.8.0. As mentioned, either ET or BT is displayed but not both. 
Some things to note:

- Using a Bella EVO 1 Roaster with USB Connection (modbus)
- I have verified in the machine that the ET and BT modbus device ID is correct. 1 for BT and 2 for ET
- I have installed the mac pl2303 serial driver from prolific
- same settings work fine on windows 11 pc
- I copied the windows artisan config and imported it on mac 

I'm a little stumped here and I'm not sure if it's a machine-specific issue or a bug in artisan. I believe I have a similar experience with #843 .

Appreciate any help. Thanks!

**Comment by Zero518** [comment]
<img width="1293" alt="Screenshot 2022-12-20 at 5 33 18 PM" src="https://user-images.githubusercontent.com/8577225/208633334-ff65acea-18a9-4f6e-a08f-65f96edd2cc5.png">
![IMG_3614](https://user-images.githubusercontent.com/8577225/208633445-44d85942-e231-41cd-9918-775d269d56d9.jpeg)

**Comment by MAKOMO** [comment]
Could you please do two things

1. attach here your artisan.log file. For this, click on the plus icon in the upper left corner of the Artisan window while holding the OPTION and COMMAND keys. This will open an email message with the file attached. Rename this file into artisan.log.txt and drag-and-drop it here on a message field on this thread on GitHub
2. download the current continuous build from [here](https://github.com/artisan-roaster-scope/artisan/releases/tag/continuous) and test it this works any better. Before testing, click on the plus icon while holding the OPTION and COMMAND keys to toggle the DEBUG logging ON. Run your test and attach the resulting log file as in step 1.

Thanks!

**Reply by MAKOMO** [reply]
Oh, please also attach your settings file here (save via menu Help >> Save Settings and rename it into artisan-settings.aset.txt).

**Reply by Zero518** [reply]
Attachment 1: 
[artisan.log.txt](https://github.com/artisan-roaster-scope/artisan/files/10268257/artisan.log.txt)

**Reply by Zero518** [reply]
Attachment 2 (Continuous Build):
[artisan.log.txt](https://github.com/artisan-roaster-scope/artisan/files/10268297/artisan.log.txt)

**Reply by Zero518** [reply]
Settings file:
[artisan-settings.aset.txt](https://github.com/artisan-roaster-scope/artisan/files/10268301/artisan-settings.aset.txt)


Also this happened when I ran a test for the first attachment. The ET showed briefly for about a second after i pressed START. Then it disappeared.

**Reply by MAKOMO** [reply]
These are all artisan .alog profiles you attached, but not the Artisan app log file I asked for. Could you please follow my instructions above?

**Reply by Zero518** [reply]
Apologies, my mistake. Here's the 1st log:

[artisan.log.txt](https://github.com/artisan-roaster-scope/artisan/files/10268467/artisan.log.txt)

2nd attachment (Continuous Build): 
[artisan.log.txt](https://github.com/artisan-roaster-scope/artisan/files/10268494/artisan.log.txt)

**Comment by MAKOMO** [comment, marked as answer]
I do not have many good ideas on this one. The machine just does not return an ET reading when requested. Could be a timing issue. You could try to add a delay between successive requests send by Artisan and ask Artisan to wait longer for a reply by changing the following settings in menu Config >> Ports, 3rd tab MODBUS:

- incl. serial timeout to 0.6sec
- turn off the optimizer which should be not effective in this case anyhow
- add a serial delay between commands of 30ms

<img width="1114" alt="Screenshot 2022-12-20 at 15 25 19" src="https://user-images.githubusercontent.com/13909011/208690101-26da2783-dd82-4701-8065-0b44c91e95ea.png">

**Reply by Zero518** [reply]
Tried your suggestions and I believe it worked! Will try using it 

<img width="1265" alt="Screenshot 2022-12-21 at 12 13 34 PM" src="https://user-images.githubusercontent.com/8577225/208819797-0327e4f6-bb66-4a4c-a23b-34da0f2aedc5.png">

**Reply by MAKOMO** [reply]
Excellent! Please report if this is a stable solution.

**Reply by Zero518** [reply]
Was able to roast 4 batches straight with no issues. Data is being logged properly so I think this is stable for regular use. Thanks a lot!

**Reply by MAKOMO** [reply]
Thanks for this confirmation! Happy roasting.
